sklearn/preprocessing/imputation.py

```diff
@@ -311,34 +325,50 @@ def _dense_fit(self, X, strategy, missing_values, axis):
 
             return most_frequent
 
-    def _sparse_transform(self, X, valid_stats, valid_idx):
+    def _sparse_transform(self, X, statistics):
         """transformer on sparse data."""
+        imputer_mask_ = _get_mask(X.data, self.missing_values)
+        mask_matrix_ = X.__class__((imputer_mask_, X.indices.copy(),
+                                    X.indptr.copy()), shape=X.shape,
+                                   dtype=X.dtype)
+        X, valid_stats = self._preprocess(X, statistics)
+
+        if self.axis == 0:
+            all_missing_ = np.where(np.all(mask_matrix_.A, axis=0))[0]
```

Comment:
yeah sorry I understand the memory wastage. How about something like ``all_missing = np.where(mask_matrix.sum(axis=0).A == X.shape[0])[0]`` ?